runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientFetchFactory.java

```diff
@@ -1949,6 +1948,7 @@ private void doApplicationData(
             OctetsFW payload,
             Flyweight extension)
         {
+            flushOrDataFramesSent++;
```

Comment:
Please move this to the end of the method, so it represents what has already happened.